File: clock/clock.go
package clock

import "fmt"

type Timer interface {
	Equal(t Timer) bool
	Add(minute int)
	Sub(minute int)
	Hour() int
	Minute() int
	String() string
}

type timerImpl struct {
	minute int
}

func New() Timer {
	return &timerImpl{minute: 0}
}

func (t *timerImpl) Equal(another Timer) bool {
	return t.Hour() == another.Hour() && t.Minute() == another.Minute()
}

func (t *timerImpl) Add(minute int) {
	t.minute += minute
	t.minute %= 60 * 24
}

func (t *timerImpl) Sub(minute int) {
	t.minute -= minute
	for {
		if t.minute < 0 {
			t.minute += 60 * 24
		} else {
			break
		}
	}
}

func (t *timerImpl) Hour() int {
	return t.minute / 60
}

func (t *timerImpl) Minute() int {
	return t.minute % 60
}

func (t *timerImpl) String() string {
	return fmt.Sprintf("%02d:%02d", t.Hour(), t.Minute())
}

File: src/clock.rs
use std::fmt::Debug;
use std::fmt::Display;
use std::fmt::Formatter;
use std::fmt::Result;

pub struct Timer {
    minute: i64,
}

impl Timer {
    pub fn new() -> Timer {
        return Timer { minute: 0 };
    }

    pub fn hour(&self) -> i64 {
        return self.minute / 60;
    }

    pub fn minute(&self) -> i64 {
        return self.minute % 60;
    }

    pub fn add(&mut self, minute: i64){
        self.minute += minute;
        self.minute %= 60 * 24;
    }
}

impl Display for Timer {
    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
        let h = self.hour();
        let m = self.minute();
        return write!(f, "{:>02}:{:>02}", h, m);
    }
}

impl PartialEq for Timer {
    fn eq(&self, another: &Timer) -> bool {
        return self.hour() == another.hour() && self.minute() == another.minute();
    }
}

impl Debug for Timer {
    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
        return write!(f, "{:>02}:{:>02}", self.hour(), self.minute());
    }
}

#[test]
fn test_equal(){
    let mut t0 = Timer::new();
    let mut t1 = Timer::new();
    t0.add(1000);
    t1.add(300);
    assert_ne!(t0, t1);
    t1.add(700);
    assert_eq!(t0, t1);
}